packages/fxa-auth-server/lib/server.js

```diff
@@ -124,6 +124,36 @@ async function create (log, error, config, routes, db, oauthdb, translator) {
             return null;
           }
           return token;
+        })
+        .then((sessionToken) => {
+          if (! requireMaximumAssuranceLevel) {
+            return sessionToken;
+          }
+
+          return db.totpToken(sessionToken.uid)
+            .then((totpToken) => {
+              if (totpToken && totpToken.verified && totpToken.enabled) {
+                if (sessionToken.authenticatorAssuranceLevel >= 2) {
+                  return sessionToken;
+                }
+
+                // Session tokens created before a TOTP token became verified
+                // are considered maximum verified.
+                if (totpToken.createdAt > sessionToken.createdAt && ! sessionToken.tokenVerificationId) {
```

Comment:
Should you be checking against `sessionToken.verifiedAt` here rather than `sessionToken.createdAt`?